### ISSUE TITLE
Not all scenarios covered in object caching

### ISSUE THREAD

**Issue by chipsenkbeil** [ensime/ensime-server#1505]
As discussed with @dickwall, I have missed a couple of locations where object references should be cached to allow correct value inspection.

Specifically, the following places need caching:
- https://github.com/ensime/ensime-server/blob/1.0/core/src/main/scala/org/ensime/core/debug/DebugActor.scala#L334

    ```scala
    val result = s.tryThread(threadId).map(_.cache()).flatMap(t =>
          suspendAndExecute(t, action(s, t)))
    ```

- https://github.com/ensime/ensime-server/blob/1.0/core/src/main/scala/org/ensime/core/debug/DebugActor.scala#L249

    ```scala
    s.tryThread(threadId.id).map(_.cache()) match { // ...
    ```

Furthermore, in the case of accessing a thread that has not been cached when issued a stack slot request, we should search for the thread if not found in the cache:

https://github.com/ensime/ensime-server/blob/1.0/core/src/main/scala/org/ensime/core/debug/DebugActor.scala#L391

```scala
objectCache.load(threadId.id)
  .orElse(scalaVirtualMachine.tryThread(threadId.id).toOption).flatMap { // ...
```

The above requires making the scala virtual machine instance available to the `lookupValue` function.

There may also be additional places that I missed with the caching invocation.

Furthermore, the cache method currently returns an object info wrapper. This means certain code will require a hard cast (such as to a thread info, etc). Debugger api 1.1.0-M2 should include a fix to return the actual type instead of the base type of object.

**Comment by chipsenkbeil** [comment]
Added several more caching invocations. Also updated the debugger api to return the actual type as opposed to `ObjectInfoProfile` to allow smoother usage. Will need to check with @dickwall later to see if I got all of the cases (not quite ready for him to try it).

**Comment by chipsenkbeil** [comment]
Resolver didn't seem to work in the PR. Fixed via #1527.
